src/Service.GraphQLBuilder/GraphQLStoredProcedureBuilder.cs

```diff
@@ -37,24 +38,30 @@ public static class GraphQLStoredProcedureBuilder
             // which are needed because parameter and column names can differ.
             StoredProcedureDefinition spdef = (StoredProcedureDefinition)dbObject.SourceDefinition;
 
-            // Create input value definitions from parameters defined in runtime config.
-            if (entity.Source.Parameters is not null)
+            // Create input value definitions from parameters defined in stored proc.
+            if (spdef is not null)
             {
-                foreach (string param in entity.Source.Parameters.Keys)
+                foreach ((string param, ParameterDefinition definition) in spdef.Parameters)
```

Comment:
So by iterating over the resolved Stored procedure definition retreived from the db, you can handle the case when the customer provides default values for the input param in the runtime config that dont match whats there in the stored proc?

Reply:
This is already validated while creating the stored procedure definition here: 
<img width="907" alt="image" src="https://github.com/Azure/data-api-builder/assets/124841904/9473acb6-3949-4f10-99ae-54cd43a10749">